lib/inspec/rule.rb

```diff
@@ -46,12 +46,11 @@ def initialize(id, profile_id, resource_dsl, opts, &block)
       return unless block_given?
 
       begin
-        instance_eval(&block)
-
-        # By applying waivers *after* the instance eval, we assure that
-        # waivers have higher precedence than only_if.
```

Comment:
Not saw any impact on the precedence even if _apply_waivers is called before instance_eval. Calling _apply_waivers here is required to get all the waivers data so that only in valid condition we will evaluate the describe block like for example if the run is set to true and the if the waiver is expired.